dialoptions.go

```diff
@@ -391,6 +391,11 @@ func WithDialer(f func(string, time.Duration) (net.Conn, error)) DialOption {
 // all the RPCs and underlying network connections in this ClientConn.
 func WithStatsHandler(h stats.Handler) DialOption {
 	return newFuncDialOption(func(o *dialOptions) {
+		if h == nil {
+			// Do not allow a nil stats handler, which would otherwise cause
+			// panics.
```

Comment:
Should we throw a warning log or something to let the user know about this instead of silently ignoring it.

Reply:
Good call!  It's a usage error, but we have no way of returning said error, so logging at error level seems appropriate.  Done.